bin/parse_package_resolved

```diff
@@ -10,23 +10,29 @@ rescue StandardError
   exit(1)
 end
 
-resolved_object = nil
-begin
-  resolved_object = JSON.parse(package_resolved_text)
-rescue JSON::ParserError
-  warn 'Unable to parse Package.resolved JSON'
-  exit(1)
+resolved_object = begin
+  JSON.parse(package_resolved_text)
+                  rescue JSON::ParserError
+                    warn 'Unable to parse Package.resolved JSON'
+                    exit(1)
 end
 
-pinned_dependencies = resolved_object['object']['pins']
+pinned_dependencies = if resolved_object.key?('object') && resolved_object['object'].key?('pins')
```

Comment:
minor nit: this can be updated to something like
```ruby
if resolved_object.dig('object', 'pin')
```

See https://apidock.com/ruby/Hash/dig